[Question]: Gazebo simulation error

### Question

Hi,
When I launch gazebo simulation command, following error shows up. An Empty Gazebo simulation(with trosenrobtics office setup -walls) and rviz opens up. Why is it so? below is lg of commands. 

locobot@locobot:~/interbotix_ws$ ros2 launch interbotix_xslocobot_sim xslocobot_gz_classic.launch.py robot_model:=locobot_wx250s robot_description:=gz_classic
[INFO] [launch]: All log files can be found below /home/locobot/.ros/log/2025-01-22-06-43-47-988020-locobot-22853
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [gzserver-1]: process started with pid [22862]
[INFO] [gzclient   -2]: process started with pid [22864]
[INFO] [spawn_entity.py-3]: process started with pid [22867]
[INFO] [robot_state_publisher-4]: process started with pid [22870]
[INFO] [rviz2-5]: process started with pid [22872]
[ERROR] [robot_state_publisher-4]: process has died [pid 22870, exit code -6, cmd '/opt/ros/galactic/lib/robot_state_publisher/robot_state_publisher --ros-args -r __ns:=/locobot --params-file /tmp/launch_params_tzb47c5k'].

### Robot Model

locobot 250s

### Operating System

Ubuntu 20.04

### ROS Version

ROS 2 Galactic

### Additional Info

_No response_

Hi, 

Gazebo simulatation did open along with locobot_250s on locobot. I had to use below command
ros2 launch interbotix_xslocobot_moveit xslocobot_moveit.launch.py robot_model:=locobot_wx250s use_lidar:=true hardware_type:=gz_classic

However, after I applied force on one link, gz sim got stuck and it started to give errors like below : 
```
locobot@locobot:~$ ros2 launch interbotix_xslocobot_moveit xslocobot_moveit.launch.py robot_model:=locobot_wx250s use_lidar:=true hardware_type:=gz_classic
[INFO] [launch]: All log files can be found below /home/locobot/.ros/log/2025-01-28-07-28-48-936864-locobot-6717
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [move_group-1]: process started with pid [6731]
[INFO] [rviz2-2]: process started with pid [6733]
[INFO] [gzserver-3]: process started with pid [6735]
[INFO] [gzclient   -4]: process started with pid [6737]
[INFO] [spawn_entity.py-5]: process started with pid [6740]
[INFO] [robot_state_publisher-6]: process started with pid [6743]
[move_group-1] [WARN] [1738070931.268481076] [move_group.move_group]: MoveGroup launched without ~default_planning_pipeline specifying the namespace for the default planning pipeline configuration
[move_group-1] [WARN] [1738070931.269058319] [move_group.move_group]: Falling back to using the the move_group node namespace (deprecated behavior).
[move_group-1] Error:   Link 'ee_gripper_link' specified as parent for end effector 'interbotix_gripper' is not known to the URDF
[move_group-1]          at line 423 in /tmp/binarydeb/ros-galactic-srdfdom-2.0.3/src/model.cpp
[move_group-1] Warning: Link 'locobot/forearm_link' is not known to URDF. Cannot disable collisons.
[move_group-1]          at line 555 in /tmp/binarydeb/ros-galactic-srdfdom-2.0.3/src/model.cpp
[move_group-1] Warning: Link 'locobot/forearm_link' is not known to URDF. Cannot disable collisons.
[move_group-1]          at line 555 in /tmp/binarydeb/ros-galactic-srdfdom-2.0.3/src/model.cpp
[move_group-1] Warning: Link 'locobot/forearm_link' is not known to URDF. Cannot disable collisons.
[move_group-1]          at line 550 in /tmp/binarydeb/ros-galactic-srdfdom-2.0.3/src/model.cpp
[move_group-1] Warning: Link 'locobot/forearm_link' is not known to URDF. Cannot disable collisons.
[move_group-1]          at line 550 in /tmp/binarydeb/ros-galactic-srdfdom-2.0.3/src/model.cpp
[move_group-1] [INFO] [1738070931.276743422] [moveit_rdf_loader.rdf_loader]: Loaded robot model in 0 seconds
[move_group-1] [INFO] [1738070931.276795657] [moveit_robot_model.robot_model]: Loading robot model 'locobot'...
[move_group-1] [INFO] [1738070931.276814109] [moveit_robot_model.robot_model]: No root/virtual joint specified in SRDF. Assuming fixed joint
[move_group-1] The root link locobot/base_link has an inertia specified in the URDF, but KDL does not support a root link with an inertia.  As a workaround, you can add an extra dummy link to your URDF.
[move_group-1] Link locobot/bump_caster_link had 0 children
[move_group-1] Link locobot/bump_front_center had 0 children
[move_group-1] Link locobot/bump_front_left had 0 children
[move_group-1] Link locobot/bump_front_right had 0 children
[move_group-1] Link locobot/bump_left had 0 children
[move_group-1] Link locobot/bump_right had 0 children
[move_group-1] Link locobot/button_1 had 0 children
[move_group-1] Link locobot/button_2 had 0 children
[move_group-1] Link locobot/button_power had 0 children
[move_group-1] Link locobot/camera_tower_link had 2 children
[move_group-1] Link locobot/lidar_tower_link had 1 children
[move_group-1] Link locobot/laser_frame_link had 0 children
[move_group-1] Link locobot/pan_link had 1 children
[move_group-1] Link locobot/tilt_link had 1 children
[move_group-1] Link camera_locobot_link had 1 children
[move_group-1] Link locobot/camera_depth_link had 0 children
[move_group-1] Link cliff_front_left had 0 children
[move_group-1] Link cliff_front_right had 0 children
[move_group-1] Link cliff_side_left had 0 children
[move_group-1] Link cliff_side_right had 0 children
[move_group-1] Link locobot/front_caster_link had 0 children
[move_group-1] Link locobot/imu_link had 0 children
[move_group-1] Link locobot/ir_omni had 0 children
[move_group-1] Link locobot/ir_intensity_front_center_left had 0 children
[move_group-1] Link locobot/ir_intensity_front_center_right had 0 children
[move_group-1] Link locobot/ir_intensity_front_left had 0 children
[move_group-1] Link locobot/ir_intensity_front_right had 0 children
[move_group-1] Link locobot/ir_intensity_left had 0 children
[move_group-1] Link locobot/ir_intensity_right had 0 children
[move_group-1] Link locobot/ir_intensity_side_left had 0 children
[move_group-1] Link locobot/mouse had 0 children
[move_group-1] Link locobot/plate_link had 2 children
[move_group-1] Link locobot/arm_stand_link had 1 children
[move_group-1] Link locobot/arm_base_link had 1 children
[move_group-1] Link locobot/shoulder_link had 1 children
[move_group-1] Link locobot/upper_arm_link had 1 children
[move_group-1] Link locobot/upper_forearm_link had 1 children
[move_group-1] Link locobot/lower_forearm_link had 1 children
[move_group-1] Link locobot/wrist_link had 1 children
[move_group-1] Link locobot/gripper_link had 1 children
[move_group-1] Link locobot/ee_arm_link had 3 children
[move_group-1] Link locobot/ar_tag_link had 0 children
[move_group-1] Link locobot/gripper_prop_link had 0 children
[move_group-1] Link locobot/gripper_bar_link had 1 children
[move_group-1] Link locobot/fingers_link had 3 children
[move_group-1] Link locobot/ee_gripper_link had 0 children
[move_group-1] Link locobot/left_finger_link had 0 children
[move_group-1] Link locobot/right_finger_link had 0 children
[move_group-1] Link locobot/battery_link had 0 children
[move_group-1] Link locobot/wheel_drop_left had 1 children
[move_group-1] Link locobot/left_wheel had 0 children
[move_group-1] Link locobot/wheel_drop_right had 1 children
[move_group-1] Link locobot/right_wheel had 0 children
[move_group-1] [INFO] [1738070931.736517213] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Publishing maintained planning scene on 'monitored_planning_scene'
[move_group-1] [INFO] [1738070931.736884337] [moveit.ros_planning_interface.moveit_cpp]: Listening to 'locobot/joint_states' for joint states
[move_group-1] [INFO] [1738070931.737776719] [moveit_ros.current_state_monitor]: Listening to joint states on topic 'locobot/joint_states'
[move_group-1] [INFO] [1738070931.738325041] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Listening to '/attached_collision_object' for attached collision objects
[move_group-1] [INFO] [1738070931.738348461] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Starting planning scene monitor
[move_group-1] [INFO] [1738070931.738740666] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Listening to '/planning_scene'
[move_group-1] [INFO] [1738070931.738761891] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Starting world geometry update monitor for collision objects, attached objects, octomap updates.
[move_group-1] [INFO] [1738070931.739166058] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Listening to 'collision_object'
[move_group-1] [INFO] [1738070931.739517607] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Listening to 'planning_scene_world' for planning scene world geometry
[move_group-1] [WARN] [1738070931.740066697] [moveit.ros.occupancy_map_monitor.middleware_handle]: Resolution not specified for Octomap. Assuming resolution = 0.1 instead
[move_group-1] [ERROR] [1738070931.740098296] [moveit.ros.occupancy_map_monitor.middleware_handle]: No sensor plugin specified for octomap updater ; ignoring.
[move_group-1] [INFO] [1738070931.740115682] [moveit.ros.occupancy_map_monitor.middleware_handle]: Skipping octomap updater plugin ''
[ERROR] [gzserver-3]: process has died [pid 6735, exit code 255, cmd 'gzserver /home/locobot/interbotix_ws/install/interbotix_common_sim/share/interbotix_common_sim/worlds/interbotix.world                                                                      -s libgazebo_ros_init.so   -s libgazebo_ros_factory.so   -s libgazebo_ros_force_system.so       '].
[move_group-1] [INFO] [1738070932.037468600] [moveit.ros_planning_interface.moveit_cpp]: Loading planning pipeline 'move_group'
[move_group-1] [INFO] [1738070932.060398107] [moveit.ros_planning.planning_pipeline]: Using planning interface 'OMPL'
[move_group-1] [INFO] [1738070932.063058791] [moveit_ros.add_time_optimal_parameterization]: Param 'move_group.path_tolerance' was not set. Using default value: 0.100000
[move_group-1] [INFO] [1738070932.063073759] [moveit_ros.add_time_optimal_parameterization]: Param 'move_group.resample_dt' was not set. Using default value: 0.100000
[move_group-1] [INFO] [1738070932.063078013] [moveit_ros.add_time_optimal_parameterization]: Param 'move_group.min_angle_change' was not set. Using default value: 0.001000
[move_group-1] [INFO] [1738070932.063091348] [moveit_ros.fix_workspace_bounds]: Param 'move_group.default_workspace_bounds' was not set. Using default value: 10.000000
[move_group-1] [INFO] [1738070932.063103727] [moveit_ros.fix_start_state_bounds]: Param 'move_group.start_state_max_bounds_error' was set to 0.100000
[move_group-1] [INFO] [1738070932.063108369] [moveit_ros.fix_start_state_bounds]: Param 'move_group.start_state_max_dt' was not set. Using default value: 0.500000
[move_group-1] [INFO] [1738070932.063118488] [moveit_ros.fix_start_state_collision]: Param 'move_group.start_state_max_dt' was not set. Using default value: 0.500000
[move_group-1] [INFO] [1738070932.063122954] [moveit_ros.fix_start_state_collision]: Param 'move_group.jiggle_fraction' was not set. Using default value: 0.020000
[move_group-1] [INFO] [1738070932.063133223] [moveit_ros.fix_start_state_collision]: Param 'move_group.max_sampling_attempts' was not set. Using default value: 100
[move_group-1] [INFO] [1738070932.063143638] [moveit.ros_planning.planning_pipeline]: Using planning request adapter 'Add Time Optimal Parameterization'
[move_group-1] [INFO] [1738070932.063148326] [moveit.ros_planning.planning_pipeline]: Using planning request adapter 'Fix Workspace Bounds'
[move_group-1] [INFO] [1738070932.063172058] [moveit.ros_planning.planning_pipeline]: Using planning request adapter 'Fix Start State Bounds'
[move_group-1] [INFO] [1738070932.063175429] [moveit.ros_planning.planning_pipeline]: Using planning request adapter 'Fix Start State In Collision'
[move_group-1] [INFO] [1738070932.063178016] [moveit.ros_planning.planning_pipeline]: Using planning request adapter 'Fix Start State Path Constraints'
[move_group-1] [INFO] [1738070932.084739211] [moveit.plugins.moveit_simple_controller_manager]: Added FollowJointTrajectory controller for locobot/arm_controller
[move_group-1] [INFO] [1738070932.086296775] [moveit.plugins.moveit_simple_controller_manager]: Added FollowJointTrajectory controller for locobot/gripper_controller
[move_group-1] [INFO] [1738070932.086453968] [moveit.plugins.moveit_simple_controller_manager]: Returned 2 controllers in list
[move_group-1] [INFO] [1738070932.087099631] [moveit_ros.trajectory_execution_manager]: Trajectory execution is managing controllers
[move_group-1] [INFO] [1738070932.087120073] [move_group.move_group]: MoveGroup debug mode is ON
[move_group-1] [INFO] [1738070932.105556134] [move_group.move_group]: 
[move_group-1] 
[move_group-1] ********************************************************
[move_group-1] * MoveGroup using: 
[move_group-1] *     - ApplyPlanningSceneService
[move_group-1] *     - ClearOctomapService
[move_group-1] *     - CartesianPathService
[move_group-1] *     - ExecuteTrajectoryAction
[move_group-1] *     - GetPlanningSceneService
[move_group-1] *     - KinematicsService
[move_group-1] *     - MoveAction
[move_group-1] *     - MotionPlanService
[move_group-1] *     - QueryPlannersService
[move_group-1] *     - StateValidationService
[move_group-1] ********************************************************
[move_group-1] 
[move_group-1] [INFO] [1738070932.107171842] [moveit_move_group_capabilities_base.move_group_context]: MoveGroup context using planning plugin ompl_interface/OMPLPlanner
[move_group-1] [INFO] [1738070932.107614494] [moveit_move_group_capabilities_base.move_group_context]: MoveGroup context initialization complete
[move_group-1] Loading 'move_group/ApplyPlanningSceneService'...
[move_group-1] Loading 'move_group/ClearOctomapService'...
[move_group-1] Loading 'move_group/MoveGroupCartesianPathService'...
[move_group-1] Loading 'move_group/MoveGroupExecuteTrajectoryAction'...
[move_group-1] Loading 'move_group/MoveGroupGetPlanningSceneService'...
[move_group-1] Loading 'move_group/MoveGroupKinematicsService'...
[move_group-1] Loading 'move_group/MoveGroupMoveAction'...
[move_group-1] Loading 'move_group/MoveGroupPlanService'...
[move_group-1] Loading 'move_group/MoveGroupQueryPlannersService'...
[move_group-1] Loading 'move_group/MoveGroupStateValidationService'...
[move_group-1] 
[move_group-1] You can start planning now!
[move_group-1] 
[ERROR] [spawn_entity.py-5]: process has died [pid 6740, exit code 1, cmd '/opt/ros/galactic/lib/gazebo_ros/spawn_entity.py -entity robot_description -topic robot_description -x 0.0 -y 0.0 -z 0.0 -Y 0.0 --ros-args -r __node:=spawn_robot -r __ns:=/locobot'].
[INFO] [spawner-7]: process started with pid [6912]
[spawner-7] [INFO] [1738070938.141581320] [joint_state_broadcaster_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-7] [INFO] [1738070940.155714835] [joint_state_broadcaster_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-7] [INFO] [1738070942.169685489] [joint_state_broadcaster_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-7] [INFO] [1738070944.178694147] [joint_state_broadcaster_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-7] [INFO] [1738070946.188915163] [joint_state_broadcaster_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-7] [ERROR] [1738070948.199540734] [joint_state_broadcaster_spawner]: Controller manager not available
[ERROR] [spawner-7]: process has died [pid 6912, exit code 1, cmd '/opt/ros/galactic/lib/controller_manager/spawner -c locobot/controller_manager joint_state_broadcaster --ros-args -r __node:=joint_state_broadcaster_spawner'].
[INFO] [spawner-8]: process started with pid [6979]
[INFO] [spawner-9]: process started with pid [6981]
[INFO] [spawner-10]: process started with pid [6983]
[INFO] [spawner-11]: process started with pid [6985]
[spawner-8] [INFO] [1738070949.106149085] [gripper_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-11] [INFO] [1738070949.163379788] [diffdrive_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-10] [INFO] [1738070949.199754302] [camera_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-9] [INFO] [1738070949.222687830] [arm_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-8] [INFO] [1738070951.117513226] [gripper_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-11] [INFO] [1738070951.173075197] [diffdrive_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-10] [INFO] [1738070951.209489565] [camera_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-9] [INFO] [1738070951.236920447] [arm_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-8] [INFO] [1738070953.130409388] [gripper_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-11] [INFO] [1738070953.184704008] [diffdrive_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-10] [INFO] [1738070953.219422636] [camera_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-9] [INFO] [1738070953.248518840] [arm_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-8] [INFO] [1738070955.140297792] [gripper_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-11] [INFO] [1738070955.195793107] [diffdrive_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-10] [INFO] [1738070955.230130367] [camera_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-9] [INFO] [1738070955.259853627] [arm_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
[spawner-8] [INFO] [1738070957.152806068] [gripper_controller_spawner]: Waiting for '/locobot/controller_manager' node to exist
^C[WARNING] [launch]: user interrupted with ctrl-c (SIGINT)
[move_group-1] [INFO] [1738070957.163148055] [rclcpp]: signal_handler(signal_value=2)
[spawner-10] Traceback (most recent call last):
[spawner-10]   File "/opt/ros/galactic/lib/controller_manager/spawner", line 20, in <module>
[spawner-10]     ret = main()
[spawner-10]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 162, in main
[spawner-10]     if not wait_for_controller_manager(node, controller_manager_name,
[spawner-10]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 95, in wait_for_controller_manager
[spawner-10]     node_and_namespace = wait_for_value_or(
[spawner-10]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 57, in wait_for_value_or
[spawner-10]     time.sleep(0.2)
[spawner-10] KeyboardInterrupt
[spawner-8] Traceback (most recent call last):
[spawner-8]   File "/opt/ros/galactic/lib/controller_manager/spawner", line 20, in <module>
[spawner-8]     ret = main()
[spawner-8]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 162, in main
[spawner-8]     if not wait_for_controller_manager(node, controller_manager_name,
[spawner-8]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 95, in wait_for_controller_manager
[spawner-8]     node_and_namespace = wait_for_value_or(
[spawner-8]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 57, in wait_for_value_or
[spawner-8]     time.sleep(0.2)
[spawner-8] KeyboardInterrupt
[spawner-11] Traceback (most recent call last):
[spawner-11]   File "/opt/ros/galactic/lib/controller_manager/spawner", line 20, in <module>
[spawner-11]     ret = main()
[spawner-11]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 162, in main
[spawner-11]     if not wait_for_controller_manager(node, controller_manager_name,
[spawner-11]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 95, in wait_for_controller_manager
[spawner-11]     node_and_namespace = wait_for_value_or(
[spawner-11]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 57, in wait_for_value_or
[spawner-11]     time.sleep(0.2)
[spawner-11] KeyboardInterrupt
[spawner-9] Traceback (most recent call last):
[spawner-9]   File "/opt/ros/galactic/lib/controller_manager/spawner", line 20, in <module>
[spawner-9]     ret = main()
[spawner-9]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 162, in main
[spawner-9]     if not wait_for_controller_manager(node, controller_manager_name,
[spawner-9]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 95, in wait_for_controller_manager
[spawner-9]     node_and_namespace = wait_for_value_or(
[spawner-9]   File "/opt/ros/galactic/lib/controller_manager/spawner.py", line 57, in wait_for_value_or
[spawner-9]     time.sleep(0.2)
[spawner-9] KeyboardInterrupt
[move_group-1] [INFO] [1738070957.218931802] [moveit.ros_planning_interface.moveit_cpp]: Deleting MoveItCpp
[move_group-1] [INFO] [1738070957.221112583] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Stopped publishing maintained planning scene.
[move_group-1] [INFO] [1738070957.222837310] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Stopping world geometry monitor
[move_group-1] [INFO] [1738070957.223724086] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Stopping planning scene monitor
[ERROR] [spawner-10]: process has died [pid 6983, exit code -2, cmd '/opt/ros/galactic/lib/controller_manager/spawner -c locobot/controller_manager camera_controller --ros-args -r __node:=camera_controller_spawner'].
[ERROR] [spawner-8]: process has died [pid 6979, exit code -2, cmd '/opt/ros/galactic/lib/controller_manager/spawner -c locobot/controller_manager gripper_controller --ros-args -r __node:=gripper_controller_spawner'].
[move_group-1] Warning: class_loader.ClassLoader: SEVERE WARNING!!! Attempting to unload library while objects created by this loader exist in the heap! You should delete your objects before attempting to unload the library or destroying the ClassLoader. The library will NOT be unloaded.
[move_group-1]          at line 127 in /tmp/binarydeb/ros-galactic-class-loader-2.1.2/src/class_loader.cpp
[ERROR] [spawner-9]: process has died [pid 6981, exit code -2, cmd '/opt/ros/galactic/lib/controller_manager/spawner -c locobot/controller_manager arm_controller --ros-args -r __node:=arm_controller_spawner'].
[ERROR] [spawner-11]: process has died [pid 6985, exit code -2, cmd '/opt/ros/galactic/lib/controller_manager/spawner -c locobot/controller_manager diffdrive_controller --ros-args -r __node:=diffdrive_controller_spawner'].
[INFO] [robot_state_publisher-6]: process has finished cleanly [pid 6743]
[ERROR] [gzclient   -4]: process has died [pid 6737, exit code -2, cmd 'gzclient   '].
[INFO] [move_group-1]: process has finished cleanly [pid 6731]
[move_group-1] 
[INFO] [rviz2-2]: process has finished cleanly [pid 6733]
```

why are these spawner errors happening? Why did it not spawn again? 

Thanks

> ```
> [ERROR] [gzserver-3]: process has died [pid 6735, exit code 255, cmd 'gzserver /home/locobot/interbotix_ws/install/interbotix_common_sim/share/interbotix_common_sim/worlds/interbotix.world                                                                      -s libgazebo_ros_init.so   -s libgazebo_ros_factory.so   -s libgazebo_ros_force_system.so       '].
> ```

Looks like gzserver is crashing. Please re-run gazebo with verbose logging.

if I try to apply force on link without making kinematic value for individual links to true, nothing happens. So I tried setting it to true, then the robot parts broke down in the gazebo. I managed to attach screenshot just before it wend off balance. I have added log about missing joints after robot is broken. 

![Image](https://github.com/user-attachments/assets/f7816981-9118-4f94-8003-b7f6bb3f62fe)

[ERROR] [rviz2-2]: process has died [pid 31747, exit code -11, cmd '/opt/ros/galactic/lib/rviz2/rviz2 -d /home/locobot/interbotix_ws/install/interbotix_xslocobot_moveit/share/interbotix_xslocobot_moveit/rviz/xslocobot_moveit.rviz -f locobot/base_link --ros-args -r __node:=rviz2 --params-file /tmp/launch_params_cahifeje --params-file /tmp/launch_params_xzo0wiju --params-file /tmp/launch_params_mft6mdzb --params-file /home/locobot/interbotix_ws/install/interbotix_xslocobot_moveit/share/interbotix_xslocobot_moveit/config/kinematics.yaml --params-file /tmp/launch_params_rg285ep6 -r locobot/get_planning_scene:=/locobot/get_planning_scene -r /arm_controller/follow_joint_trajectory:=/locobot/arm_controller/follow_joint_trajectory -r /gripper_controller/follow_joint_trajectory:=/locobot/gripper_controller/follow_joint_trajectory'].
[move_group-1] [WARN] [1738169812.482421591] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: The complete state of the robot is not yet known.  Missing wheel_drop_left_joint, wheel_drop_right_joint
[move_group-1] [WARN] [1738169813.508674950] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: The complete state of the robot is not yet known.  Missing wheel_drop_left_joint, wheel_drop_right_joint
[move_group-1] Error:   TF_NAN_INPUT: Ignoring transform for child_frame_id "locobot/left_finger_link" from authority "Authority undetectable" because of a nan value in the transform (-nan -nan -nan) (0.000000 0.000000 0.000000 1.000000)
[move_group-1]          at line 298 in /tmp/binarydeb/ros-galactic-tf2-0.17.5/src/buffer_core.cpp
[move_group-1] Error:   TF_NAN_INPUT: Ignoring transform for child_frame_id "locobot/right_finger_link" from authority "Authority undetectable" because of a nan value in the transform (-nan -nan -nan) (0.000000 0.000000 0.000000 1.000000)
[move_group-1]          at line 298 in /tmp/binarydeb/ros-galactic-tf2-0.17.5/src/buffer_core.cpp
[move_group-1] [WARN] [1738169814.542167617] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: The complete state of the robot is not yet known.  Missing wheel_drop_left_joint, wheel_drop_right_joint

For some reason, locobot is missing again in gazebo. I ran with verbose set to true. It is again just Rviz and empty room in gazebo. below is log 

```
locobot@locobot:~$ ros2 launch interbotix_xslocobot_moveit xslocobot_moveit.launch.py robot_model:=locobot_wx250s use_lidar:=true hardware_type:=gz_classic verbose:=true
[INFO] [launch]: All log files can be found below /home/locobot/.ros/log/2025-01-30-09-25-56-576504-locobot-7914
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [move_group-1]: process started with pid [7933]
[INFO] [rviz2-2]: process started with pid [7935]
[INFO] [gzserver-3]: process started with pid [7937]
[INFO] [gzclient   -4]: process started with pid [7939]
[INFO] [spawn_entity.py-5]: process started with pid [7942]
[INFO] [robot_state_publisher-6]: process started with pid [7945]
[move_group-1] [WARN] [1738250758.868341008] [move_group.move_group]: MoveGroup launched without ~default_planning_pipeline specifying the namespace for the default planning pipeline configuration
[move_group-1] [WARN] [1738250758.869021157] [move_group.move_group]: Falling back to using the the move_group node namespace (deprecated behavior).
[move_group-1] Error:   Link 'ee_gripper_link' specified as parent for end effector 'interbotix_gripper' is not known to the URDF
[move_group-1]          at line 423 in /tmp/binarydeb/ros-galactic-srdfdom-2.0.3/src/model.cpp
[move_group-1] Warning: Link 'locobot/forearm_link' is not known to URDF. Cannot disable collisons.
[move_group-1]          at line 555 in /tmp/binarydeb/ros-galactic-srdfdom-2.0.3/src/model.cpp
[move_group-1] Warning: Link 'locobot/forearm_link' is not known to URDF. Cannot disable collisons.
[move_group-1]          at line 555 in /tmp/binarydeb/ros-galactic-srdfdom-2.0.3/src/model.cpp
[move_group-1] Warning: Link 'locobot/forearm_link' is not known to URDF. Cannot disable collisons.
[move_group-1]          at line 550 in /tmp/binarydeb/ros-galactic-srdfdom-2.0.3/src/model.cpp
[move_group-1] Warning: Link 'locobot/forearm_link' is not known to URDF. Cannot disable collisons.
[move_group-1]          at line 550 in /tmp/binarydeb/ros-galactic-srdfdom-2.0.3/src/model.cpp
[move_group-1] [INFO] [1738250758.887437747] [moveit_rdf_loader.rdf_loader]: Loaded robot model in 0 seconds
[move_group-1] [INFO] [1738250758.887489894] [moveit_robot_model.robot_model]: Loading robot model 'locobot'...
[move_group-1] [INFO] [1738250758.887501307] [moveit_robot_model.robot_model]: No root/virtual joint specified in SRDF. Assuming fixed joint
[move_group-1] The root link locobot/base_link has an inertia specified in the URDF, but KDL does not support a root link with an inertia.  As a workaround, you can add an extra dummy link to your URDF.
[move_group-1] Link locobot/bump_caster_link had 0 children
[move_group-1] Link locobot/bump_front_center had 0 children
[move_group-1] Link locobot/bump_front_left had 0 children
[move_group-1] Link locobot/bump_front_right had 0 children
[move_group-1] Link locobot/bump_left had 0 children
[move_group-1] Link locobot/bump_right had 0 children
[move_group-1] Link locobot/button_1 had 0 children
[move_group-1] Link locobot/button_2 had 0 children
[move_group-1] Link locobot/button_power had 0 children
[move_group-1] Link locobot/camera_tower_link had 2 children
[move_group-1] Link locobot/lidar_tower_link had 1 children
[move_group-1] Link locobot/laser_frame_link had 0 children
[move_group-1] Link locobot/pan_link had 1 children
[move_group-1] Link locobot/tilt_link had 1 children
[move_group-1] Link camera_locobot_link had 1 children
[move_group-1] Link locobot/camera_depth_link had 0 children
[move_group-1] Link cliff_front_left had 0 children
[move_group-1] Link cliff_front_right had 0 children
[move_group-1] Link cliff_side_left had 0 children
[move_group-1] Link cliff_side_right had 0 children
[move_group-1] Link locobot/front_caster_link had 0 children
[move_group-1] Link locobot/imu_link had 0 children
[move_group-1] Link locobot/ir_omni had 0 children
[move_group-1] Link locobot/ir_intensity_front_center_left had 0 children
[move_group-1] Link locobot/ir_intensity_front_center_right had 0 children
[move_group-1] Link locobot/ir_intensity_front_left had 0 children
[move_group-1] Link locobot/ir_intensity_front_right had 0 children
[move_group-1] Link locobot/ir_intensity_left had 0 children
[move_group-1] Link locobot/ir_intensity_right had 0 children
[move_group-1] Link locobot/ir_intensity_side_left had 0 children
[move_group-1] Link locobot/mouse had 0 children
[move_group-1] Link locobot/plate_link had 2 children
[move_group-1] Link locobot/arm_stand_link had 1 children
[move_group-1] Link locobot/arm_base_link had 1 children
[move_group-1] Link locobot/shoulder_link had 1 children
[move_group-1] Link locobot/upper_arm_link had 1 children
[move_group-1] Link locobot/upper_forearm_link had 1 children
[move_group-1] Link locobot/lower_forearm_link had 1 children
[move_group-1] Link locobot/wrist_link had 1 children
[move_group-1] Link locobot/gripper_link had 1 children
[move_group-1] Link locobot/ee_arm_link had 3 children
[move_group-1] Link locobot/ar_tag_link had 0 children
[move_group-1] Link locobot/gripper_prop_link had 0 children
[move_group-1] Link locobot/gripper_bar_link had 1 children
[move_group-1] Link locobot/fingers_link had 3 children
[move_group-1] Link locobot/ee_gripper_link had 0 children
[move_group-1] Link locobot/left_finger_link had 0 children
[move_group-1] Link locobot/right_finger_link had 0 children
[move_group-1] Link locobot/battery_link had 0 children
[move_group-1] Link locobot/wheel_drop_left had 1 children
[move_group-1] Link locobot/left_wheel had 0 children
[move_group-1] Link locobot/wheel_drop_right had 1 children
[move_group-1] Link locobot/right_wheel had 0 children
[move_group-1] [INFO] [1738250759.571771794] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Publishing maintained planning scene on 'monitored_planning_scene'
[move_group-1] [INFO] [1738250759.572578998] [moveit.ros_planning_interface.moveit_cpp]: Listening to 'locobot/joint_states' for joint states
[move_group-1] [INFO] [1738250759.573516061] [moveit_ros.current_state_monitor]: Listening to joint states on topic 'locobot/joint_states'
[move_group-1] [INFO] [1738250759.574342054] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Listening to '/attached_collision_object' for attached collision objects
[move_group-1] [INFO] [1738250759.574710433] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Starting planning scene monitor
[move_group-1] [INFO] [1738250759.575362265] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Listening to '/planning_scene'
[move_group-1] [INFO] [1738250759.575708002] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Starting world geometry update monitor for collision objects, attached objects, octomap updates.
[move_group-1] [INFO] [1738250759.576384521] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Listening to 'collision_object'
[move_group-1] [INFO] [1738250759.577038738] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Listening to 'planning_scene_world' for planning scene world geometry
[move_group-1] [WARN] [1738250759.577851544] [moveit.ros.occupancy_map_monitor.middleware_handle]: Resolution not specified for Octomap. Assuming resolution = 0.1 instead
[move_group-1] [ERROR] [1738250759.578203388] [moveit.ros.occupancy_map_monitor.middleware_handle]: No sensor plugin specified for octomap updater ; ignoring.
[move_group-1] [INFO] [1738250759.578529088] [moveit.ros.occupancy_map_monitor.middleware_handle]: Skipping octomap updater plugin ''
[move_group-1] [INFO] [1738250760.104816359] [moveit.ros_planning_interface.moveit_cpp]: Loading planning pipeline 'move_group'
[move_group-1] [INFO] [1738250760.136466897] [moveit.ros_planning.planning_pipeline]: Using planning interface 'OMPL'
[move_group-1] [INFO] [1738250760.140438788] [moveit_ros.add_time_optimal_parameterization]: Param 'move_group.path_tolerance' was not set. Using default value: 0.100000
[move_group-1] [INFO] [1738250760.140463959] [moveit_ros.add_time_optimal_parameterization]: Param 'move_group.resample_dt' was not set. Using default value: 0.100000
[move_group-1] [INFO] [1738250760.140469597] [moveit_ros.add_time_optimal_parameterization]: Param 'move_group.min_angle_change' was not set. Using default value: 0.001000
[move_group-1] [INFO] [1738250760.140487570] [moveit_ros.fix_workspace_bounds]: Param 'move_group.default_workspace_bounds' was not set. Using default value: 10.000000
[move_group-1] [INFO] [1738250760.140506106] [moveit_ros.fix_start_state_bounds]: Param 'move_group.start_state_max_bounds_error' was set to 0.100000
[move_group-1] [INFO] [1738250760.140512597] [moveit_ros.fix_start_state_bounds]: Param 'move_group.start_state_max_dt' was not set. Using default value: 0.500000
[move_group-1] [INFO] [1738250760.140527887] [moveit_ros.fix_start_state_collision]: Param 'move_group.start_state_max_dt' was not set. Using default value: 0.500000
[move_group-1] [INFO] [1738250760.140533820] [moveit_ros.fix_start_state_collision]: Param 'move_group.jiggle_fraction' was not set. Using default value: 0.020000
[move_group-1] [INFO] [1738250760.140548261] [moveit_ros.fix_start_state_collision]: Param 'move_group.max_sampling_attempts' was not set. Using default value: 100
[move_group-1] [INFO] [1738250760.140563707] [moveit.ros_planning.planning_pipeline]: Using planning request adapter 'Add Time Optimal Parameterization'
[move_group-1] [INFO] [1738250760.140570611] [moveit.ros_planning.planning_pipeline]: Using planning request adapter 'Fix Workspace Bounds'
[move_group-1] [INFO] [1738250760.140600987] [moveit.ros_planning.planning_pipeline]: Using planning request adapter 'Fix Start State Bounds'
[move_group-1] [INFO] [1738250760.140606705] [moveit.ros_planning.planning_pipeline]: Using planning request adapter 'Fix Start State In Collision'
[move_group-1] [INFO] [1738250760.140610628] [moveit.ros_planning.planning_pipeline]: Using planning request adapter 'Fix Start State Path Constraints'
[move_group-1] [INFO] [1738250760.168127866] [moveit.plugins.moveit_simple_controller_manager]: Added FollowJointTrajectory controller for locobot/arm_controller
[move_group-1] [INFO] [1738250760.169319139] [moveit.plugins.moveit_simple_controller_manager]: Added FollowJointTrajectory controller for locobot/gripper_controller
[move_group-1] [INFO] [1738250760.169812155] [moveit.plugins.moveit_simple_controller_manager]: Returned 2 controllers in list
[move_group-1] [INFO] [1738250760.170638878] [moveit_ros.trajectory_execution_manager]: Trajectory execution is managing controllers
[move_group-1] [INFO] [1738250760.170984950] [move_group.move_group]: MoveGroup debug mode is ON
[move_group-1] [INFO] [1738250760.187962216] [move_group.move_group]: 
[move_group-1] 
[move_group-1] ********************************************************
[move_group-1] * MoveGroup using: 
[move_group-1] *     - ApplyPlanningSceneService
[move_group-1] *     - ClearOctomapService
[move_group-1] *     - CartesianPathService
[move_group-1] *     - ExecuteTrajectoryAction
[move_group-1] *     - GetPlanningSceneService
[move_group-1] *     - KinematicsService
[move_group-1] *     - MoveAction
[move_group-1] *     - MotionPlanService
[move_group-1] *     - QueryPlannersService
[move_group-1] *     - StateValidationService
[move_group-1] ********************************************************
[move_group-1] 
[move_group-1] [INFO] [1738250760.189156443] [moveit_move_group_capabilities_base.move_group_context]: MoveGroup context using planning plugin ompl_interface/OMPLPlanner
[move_group-1] [INFO] [1738250760.189477441] [moveit_move_group_capabilities_base.move_group_context]: MoveGroup context initialization complete
[move_group-1] Loading 'move_group/ApplyPlanningSceneService'...
[move_group-1] Loading 'move_group/ClearOctomapService'...
[move_group-1] Loading 'move_group/MoveGroupCartesianPathService'...
[move_group-1] Loading 'move_group/MoveGroupExecuteTrajectoryAction'...
[move_group-1] Loading 'move_group/MoveGroupGetPlanningSceneService'...
[move_group-1] Loading 'move_group/MoveGroupKinematicsService'...
[move_group-1] Loading 'move_group/MoveGroupMoveAction'...
[move_group-1] Loading 'move_group/MoveGroupPlanService'...
[move_group-1] Loading 'move_group/MoveGroupQueryPlannersService'...
[move_group-1] Loading 'move_group/MoveGroupStateValidationService'...
[move_group-1] 
[move_group-1] You can start planning now!
[move_group-1] 

```

--- a/interbotix_ros_xslocobots/interbotix_xslocobot_descriptions/urdf/lidar.urdf.xacro
+++ b/interbotix_ros_xslocobots/interbotix_xslocobot_descriptions/urdf/lidar.urdf.xacro
@@ -136,7 +136,7 @@
         </ray>
         <plugin name="gazebo_ros_head_rplidar_controller" filename="libgazebo_ros_ray_sensor.so">
           <ros>
-            <remapping>~/out:=scan</remapping>
+            <remapping>~/out:=${robot_name}/scan</remapping>
           </ros>
           <output_type>sensor_msgs/LaserScan</output_type>
           <frame_name>${robot_name}/laser_frame_link</frame_name>

--- a/interbotix_ros_xslocobots/interbotix_xslocobot_descriptions/urdf/pan_and_tilt.urdf.xacro
+++ b/interbotix_ros_xslocobots/interbotix_xslocobot_descriptions/urdf/pan_and_tilt.urdf.xacro
@@ -235,27 +235,26 @@
           <ros>
             <namespace>/$(arg robot_name)</namespace>
           </ros>
-          <alwaysOn>true</alwaysOn>
-          <updateRate>0.0</updateRate>
-          <cameraName>camera</cameraName>
-          <imageTopicName>color/image_raw</imageTopicName>
-          <cameraInfoTopicName>color/camera_info</cameraInfoTopicName>
-          <depthImageTopicName>aligned_depth_to_color/image_raw</depthImageTopicName>
-          <depthImageCameraInfoTopicName>aligned_depth_to_color/camera_info</depthImageCameraInfoTopicName>
-          <pointCloudTopicName>depth_registered/points</pointCloudTopicName>
-          <frameName>${robot_name}/camera_depth_link</frameName>
-          <pointCloudCutoff>0.01</pointCloudCutoff>
-          <pointCloudCutoffMax>10</pointCloudCutoffMax>
-          <distortionK1>0.00000001</distortionK1>
-          <distortionK2>0.00000001</distortionK2>
-          <distortionK3>0.00000001</distortionK3>
-          <distortionT1>0.00000001</distortionT1>
-          <distortionT2>0.00000001</distortionT2>
-          <CxPrime>0</CxPrime>
-          <Cx>0</Cx>
-          <Cy>0</Cy>
-          <focalLength>0</focalLength>
-          <hackBaseline>0</hackBaseline>
+          <always_on>true</always_on>
+          <update_rate>0.0</update_rate>
+          <camera_name>camera</camera_name>
+          <image_topic_name>color/image_raw</image_topic_name>
+          <camera_info_topic_name>color/camera_info</camera_info_topic_name>
+          <depth_image_topic_name>aligned_depth_to_color/image_raw</depth_image_topic_name>
+          <depth_image_camera_info_topic_name>aligned_depth_to_color/camera_info</depth_image_camera_info_topic_name>
+          <point_cloud_topic_name>depth_registered/points</point_cloud_topic_name>
+          <frame_name>${robot_name}/camera_depth_link</frame_name>
+          <point_cloud_cutoff>0.01</point_cloud_cutoff>
+          <point_cloud_cutoff_max>10</point_cloud_cutoff_max>
+          <distortion_k1>0.00000001</distortion_k1>
+          <distortion_k2>0.00000001</distortion_k2>
+          <distortion_k3>0.00000001</distortion_k3>
+          <distortion_t1>0.00000001</distortion_t1>
+          <distortion_t2>0.00000001</distortion_t2>
+          <cx>0</cx>
+          <cy>0</cy>
+          <focal_length>0</focal_length>
+          <hack_baseline>0</hack_baseline>
         </plugin>
       </sensor>
     </gazebo>

--- a/interbotix_ros_xslocobots/interbotix_xslocobot_sim/rviz/xslocobot_gz_classic.rviz
+++ b/interbotix_ros_xslocobots/interbotix_xslocobot_sim/rviz/xslocobot_gz_classic.rviz
@@ -5,7 +5,7 @@ Panels:
     Property Tree Widget:
       Expanded: ~
       Splitter Ratio: 0.6176470518112183
-    Tree Height: 484
+    Tree Height: 817
 Visualization Manager:
   Class: ""
   Displays:
@@ -45,68 +45,7 @@ Visualization Manager:
         Expand Link Details: false
         Expand Tree: false
         Link Tree Style: Links in Alphabetic Order
-        base_footprint:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        base_link:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-          Value: true
-        battery_link:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-          Value: true
-        bump_caster_link:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-          Value: true
-        bump_front_center:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-          Value: true
-        bump_front_left:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        bump_front_right:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        bump_left:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        bump_right:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        button_1:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        button_2:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        button_power:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        camera_depth_link:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        camera_link:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-          Value: true
-        camera_tower_link:
+        camera_locobot_link:
           Alpha: 1
           Show Axes: false
           Show Trail: false
@@ -127,94 +66,231 @@ Visualization Manager:
           Alpha: 1
           Show Axes: false
           Show Trail: false
-        front_caster_link:
+        locobot/ar_tag_link:
           Alpha: 1
           Show Axes: false
           Show Trail: false
           Value: true
-        imu_link:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        ir_intensity_front_center_left:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        ir_intensity_front_center_right:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        ir_intensity_front_left:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        ir_intensity_front_right:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        ir_intensity_left:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        ir_intensity_right:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        ir_intensity_side_left:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        ir_omni:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        laser_frame_link:
+        locobot/arm_base_link:
           Alpha: 1
           Show Axes: false
           Show Trail: false
           Value: true
-        left_wheel:
+        locobot/arm_stand_link:
           Alpha: 1
           Show Axes: false
           Show Trail: false
           Value: true
-        lidar_tower_link:
+        locobot/base_link:
           Alpha: 1
           Show Axes: false
           Show Trail: false
           Value: true
-        mouse:
-          Alpha: 1
-          Show Axes: false
-          Show Trail: false
-        pan_link:
+        locobot/battery_link:
           Alpha: 1
           Show Axes: false
           Show Trail: false
           Value: true
-        plate_link:
+        locobot/bump_caster_link:
           Alpha: 1
           Show Axes: false
           Show Trail: false
           Value: true
-        right_wheel:
+        locobot/bump_front_center:
           Alpha: 1
           Show Axes: false
           Show Trail: false
           Value: true
-        tilt_link:
+        locobot/bump_front_left:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/bump_front_right:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/bump_left:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/bump_right:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/button_1:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/button_2:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/button_power:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/camera_depth_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/camera_tower_link:
           Alpha: 1
           Show Axes: false
           Show Trail: false
           Value: true
-        wheel_drop_left:
+        locobot/ee_arm_link:
           Alpha: 1
           Show Axes: false
           Show Trail: false
-        wheel_drop_right:
+        locobot/ee_gripper_link:
           Alpha: 1
           Show Axes: false
           Show Trail: false
+        locobot/fingers_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/front_caster_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/gripper_bar_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/gripper_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/gripper_prop_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/imu_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/ir_intensity_front_center_left:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/ir_intensity_front_center_right:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/ir_intensity_front_left:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/ir_intensity_front_right:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/ir_intensity_left:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/ir_intensity_right:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/ir_intensity_side_left:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/ir_omni:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/laser_frame_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/left_finger_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/left_wheel:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/lidar_tower_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/lower_forearm_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/mouse:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/pan_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/plate_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/right_finger_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/right_wheel:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/shoulder_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/tilt_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/upper_arm_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/upper_forearm_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+        locobot/wheel_drop_left:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/wheel_drop_right:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+        locobot/wrist_link:
+          Alpha: 1
+          Show Axes: false
+          Show Trail: false
+          Value: true
+      Mass Properties:
+        Inertia: false
+        Mass: false
       Name: RobotModel
       TF Prefix: ""
       Update Interval: 0
@@ -225,35 +301,7 @@ Visualization Manager:
       Frame Timeout: 15
       Frames:
         All Enabled: true
-        base_footprint:
-          Value: true
-        base_link:
-          Value: true
-        battery_link:
-          Value: true
-        bump_caster_link:
-          Value: true
-        bump_front_center:
-          Value: true
-        bump_front_left:
-          Value: true
-        bump_front_right:
-          Value: true
-        bump_left:
-          Value: true
-        bump_right:
-          Value: true
-        button_1:
-          Value: true
-        button_2:
-          Value: true
-        button_power:
-          Value: true
-        camera_depth_link:
-          Value: true
-        camera_link:
-          Value: true
-        camera_tower_link:
+        camera_locobot_link:
           Value: true
         cliff_front_left:
           Value: true
@@ -263,49 +311,107 @@ Visualization Manager:
           Value: true
         cliff_side_right:
           Value: true
-        front_caster_link:
+        locobot/ar_tag_link:
           Value: true
-        imu_link:
+        locobot/arm_base_link:
           Value: true
-        ir_intensity_front_center_left:
+        locobot/arm_stand_link:
           Value: true
-        ir_intensity_front_center_right:
+        locobot/base_link:
           Value: true
-        ir_intensity_front_left:
+        locobot/battery_link:
           Value: true
-        ir_intensity_front_right:
+        locobot/bump_caster_link:
           Value: true
-        ir_intensity_left:
+        locobot/bump_front_center:
           Value: true
-        ir_intensity_right:
+        locobot/bump_front_left:
           Value: true
-        ir_intensity_side_left:
+        locobot/bump_front_right:
           Value: true
-        ir_omni:
+        locobot/bump_left:
           Value: true
-        laser_frame_link:
+        locobot/bump_right:
           Value: true
-        left_wheel:
+        locobot/button_1:
           Value: true
-        lidar_tower_link:
+        locobot/button_2:
           Value: true
-        mouse:
+        locobot/button_power:
           Value: true
-        odom:
+        locobot/camera_depth_link:
           Value: true
-        pan_link:
+        locobot/camera_tower_link:
           Value: true
-        plate_link:
+        locobot/ee_arm_link:
           Value: true
-        right_wheel:
+        locobot/ee_gripper_link:
           Value: true
-        std_dock_link:
+        locobot/fingers_link:
           Value: true
-        tilt_link:
+        locobot/front_caster_link:
           Value: true
-        wheel_drop_left:
+        locobot/gripper_bar_link:
           Value: true
-        wheel_drop_right:
+        locobot/gripper_link:
+          Value: true
+        locobot/gripper_prop_link:
+          Value: true
+        locobot/imu_link:
+          Value: true
+        locobot/ir_intensity_front_center_left:
+          Value: true
+        locobot/ir_intensity_front_center_right:
+          Value: true
+        locobot/ir_intensity_front_left:
+          Value: true
+        locobot/ir_intensity_front_right:
+          Value: true
+        locobot/ir_intensity_left:
+          Value: true
+        locobot/ir_intensity_right:
+          Value: true
+        locobot/ir_intensity_side_left:
+          Value: true
+        locobot/ir_omni:
+          Value: true
+        locobot/laser_frame_link:
+          Value: true
+        locobot/left_finger_link:
+          Value: true
+        locobot/left_wheel:
+          Value: true
+        locobot/lidar_tower_link:
+          Value: true
+        locobot/locobot/base_footprint:
+          Value: true
+        locobot/locobot/odom:
+          Value: true
+        locobot/lower_forearm_link:
+          Value: true
+        locobot/mouse:
+          Value: true
+        locobot/pan_link:
+          Value: true
+        locobot/plate_link:
+          Value: true
+        locobot/right_finger_link:
+          Value: true
+        locobot/right_wheel:
+          Value: true
+        locobot/shoulder_link:
+          Value: true
+        locobot/tilt_link:
+          Value: true
+        locobot/upper_arm_link:
+          Value: true
+        locobot/upper_forearm_link:
+          Value: true
+        locobot/wheel_drop_left:
+          Value: true
+        locobot/wheel_drop_right:
+          Value: true
+        locobot/wrist_link:
           Value: true
       Marker Scale: 0.15000000596046448
       Name: TF
@@ -313,77 +419,88 @@ Visualization Manager:
       Show Axes: true
       Show Names: true
       Tree:
-        base_footprint:
-          base_link:
-            bump_caster_link:
-              {}
-            bump_front_center:
-              {}
-            bump_front_left:
-              {}
-            bump_front_right:
-              {}
-            bump_left:
-              {}
-            bump_right:
-              {}
-            button_1:
-              {}
-            button_2:
-              {}
-            button_power:
-              {}
-            camera_tower_link:
-              lidar_tower_link:
-                laser_frame_link:
-                  {}
-              pan_link:
-                tilt_link:
-                  camera_link:
-                    camera_depth_link:
-                      {}
-            cliff_front_left:
-              {}
-            cliff_front_right:
-              {}
-            cliff_side_left:
-              {}
-            cliff_side_right:
-              {}
-            front_caster_link:
-              {}
-            imu_link:
-              {}
-            ir_intensity_front_center_left:
-              {}
-            ir_intensity_front_center_right:
-              {}
-            ir_intensity_front_left:
-              {}
-            ir_intensity_front_right:
-              {}
-            ir_intensity_left:
-              {}
-            ir_intensity_right:
-              {}
-            ir_intensity_side_left:
-              {}
-            ir_omni:
-              {}
-            mouse:
-              {}
-            plate_link:
-              battery_link:
-                {}
-            wheel_drop_left:
-              left_wheel:
-                {}
-            wheel_drop_right:
-              right_wheel:
-                {}
-        odom:
-          std_dock_link:
+        locobot/base_link:
+          cliff_front_left:
             {}
+          cliff_front_right:
+            {}
+          cliff_side_left:
+            {}
+          cliff_side_right:
+            {}
+          locobot/bump_caster_link:
+            {}
+          locobot/bump_front_center:
+            {}
+          locobot/bump_front_left:
+            {}
+          locobot/bump_front_right:
+            {}
+          locobot/bump_left:
+            {}
+          locobot/bump_right:
+            {}
+          locobot/button_1:
+            {}
+          locobot/button_2:
+            {}
+          locobot/button_power:
+            {}
+          locobot/camera_tower_link:
+            locobot/lidar_tower_link:
+              locobot/laser_frame_link:
+                {}
+            locobot/pan_link:
+              locobot/tilt_link:
+                camera_locobot_link:
+                  locobot/camera_depth_link:
+                    {}
+          locobot/front_caster_link:
+            {}
+          locobot/imu_link:
+            {}
+          locobot/ir_intensity_front_center_left:
+            {}
+          locobot/ir_intensity_front_center_right:
+            {}
+          locobot/ir_intensity_front_left:
+            {}
+          locobot/ir_intensity_front_right:
+            {}
+          locobot/ir_intensity_left:
+            {}
+          locobot/ir_intensity_right:
+            {}
+          locobot/ir_intensity_side_left:
+            {}
+          locobot/ir_omni:
+            {}
+          locobot/mouse:
+            {}
+          locobot/plate_link:
+            locobot/arm_stand_link:
+              locobot/arm_base_link:
+                locobot/shoulder_link:
+                  locobot/upper_arm_link:
+                    locobot/upper_forearm_link:
+                      locobot/lower_forearm_link:
+                        locobot/wrist_link:
+                          locobot/gripper_link:
+                            locobot/ee_arm_link:
+                              locobot/ar_tag_link:
+                                {}
+                              locobot/gripper_bar_link:
+                                locobot/fingers_link:
+                                  locobot/ee_gripper_link:
+                                    {}
+                                  locobot/left_finger_link:
+                                    {}
+                                  locobot/right_finger_link:
+                                    {}
+                              locobot/gripper_prop_link:
+                                {}
+            locobot/battery_link:
+              {}
       Update Interval: 0
       Value: true
     - Class: rviz_common/Group
@@ -398,7 +515,7 @@ Visualization Manager:
           Channel Name: intensity
           Class: rviz_default_plugins/PointCloud2
           Color: 255; 255; 255
-          Color Transformer: ""
+          Color Transformer: RGB8
           Decay Time: 0
           Enabled: true
           Invert Rainbow: false
@@ -407,7 +524,7 @@ Visualization Manager:
           Min Color: 0; 0; 0
           Min Intensity: 0
           Name: PointCloud2
-          Position Transformer: ""
+          Position Transformer: XYZ
           Selectable: true
           Size (Pixels): 3
           Size (m): 0.009999999776482582
@@ -418,7 +535,7 @@ Visualization Manager:
             Filter size: 10
             History Policy: Keep Last
             Reliability Policy: Reliable
-            Value: /camera_frame_sensor/points
+            Value: /locobot/camera/points
           Use Fixed Frame: true
           Use rainbow: true
           Value: true
@@ -434,9 +551,9 @@ Visualization Manager:
             Durability Policy: Volatile
             History Policy: Keep Last
             Reliability Policy: Best Effort
-            Value: /camera_frame_sensor/image_raw
+            Value: /locobot/camera/image_raw
           Value: true
-      Enabled: true
+      Enabled: false
       Name: Perception
     - Alpha: 1
       Autocompute Intensity Bounds: true
@@ -468,14 +585,14 @@ Visualization Manager:
         Filter size: 10
         History Policy: Keep Last
         Reliability Policy: Best Effort
-        Value: /scan
+        Value: /locobot/scan
       Use Fixed Frame: true
       Use rainbow: true
       Value: false
   Enabled: true
   Global Options:
     Background Color: 0; 176; 240
-    Fixed Frame: base_footprint
+    Fixed Frame: locobot/base_link
     Frame Rate: 30
   Name: root
   Tools:
@@ -546,7 +663,7 @@ Window Geometry:
   Hide Right Dock: true
   Image:
     collapsed: false
-  QMainWindow State: 000000ff00000000fd00000004000000000000018b00000368fc020000000cfb0000001200530065006c0065006300740069006f006e00000001e10000009b0000000000000000fb0000001e0054006f006f006c002000500072006f007000650072007400690065007302000001ed000001df00000185000000a3fb000000120056006900650077007300200054006f006f02000001df000002110000018500000122fb000000200054006f006f006c002000500072006f0070006500720074006900650073003203000002880000011d000002210000017afb000000100044006900730070006c006100790073010000001600000221000000c900fffffffb0000002000730065006c0065006300740069006f006e00200062007500660066006500720200000138000000aa0000023a00000294fb00000014005700690064006500530074006500720065006f02000000e6000000d2000003ee0000030bfb0000000c004b0069006e0065006300740200000186000001060000030c00000261fb000000300049006e0074006500720062006f00740069007800200043006f006e00740072006f006c002000500061006e0065006c00000002420000013c0000000000000000fb0000000a0049006d006100670065010000023d000001410000002800fffffffb0000000c00430061006d0065007200610100000289000000f50000000000000000fb0000000a0049006d00610067006501000002b6000000c800000000000000000000000100000110000002a2fc0200000003fb0000001e0054006f006f006c002000500072006f00700065007200740069006500730100000041000000780000000000000000fb0000000a00560069006500770073000000003d000002a20000000000000000fb0000001200530065006c0065006300740069006f006e010000025a000000b200000000000000000000000200000490000000a9fc0100000001fb0000000a00560069006500770073030000004e00000080000002e10000019700000003000006080000004cfc0100000002fb0000000800540069006d00650100000000000006080000000000000000fb0000000800540069006d00650100000000000004500000000000000000000004770000036800000004000000040000000800000008fc0000000100000002000000010000000a0054006f006f006c00730000000000ffffffff0000000000000000
+  QMainWindow State: 000000ff00000000fd00000004000000000000018b0000036cfc020000000cfb0000001200530065006c0065006300740069006f006e00000001e10000009b0000000000000000fb0000001e0054006f006f006c002000500072006f007000650072007400690065007302000001ed000001df00000185000000a3fb000000120056006900650077007300200054006f006f02000001df000002110000018500000122fb000000200054006f006f006c002000500072006f0070006500720074006900650073003203000002880000011d000002210000017afb000000100044006900730070006c00610079007301000000140000036c000000c700fffffffb0000002000730065006c0065006300740069006f006e00200062007500660066006500720200000138000000aa0000023a00000294fb00000014005700690064006500530074006500720065006f02000000e6000000d2000003ee0000030bfb0000000c004b0069006e0065006300740200000186000001060000030c00000261fb000000300049006e0074006500720062006f00740069007800200043006f006e00740072006f006c002000500061006e0065006c00000002420000013c0000000000000000fb0000000a0049006d006100670065000000023e000001420000002800fffffffb0000000c00430061006d0065007200610100000289000000f50000000000000000fb0000000a0049006d00610067006501000002b6000000c800000000000000000000000100000110000002a2fc0200000003fb0000001e0054006f006f006c002000500072006f00700065007200740069006500730100000041000000780000000000000000fb0000000a00560069006500770073000000003d000002a20000000000000000fb0000001200530065006c0065006300740069006f006e010000025a000000b200000000000000000000000200000490000000a9fc0100000001fb0000000a00560069006500770073030000004e00000080000002e10000019700000003000006080000004cfc0100000002fb0000000800540069006d00650100000000000006080000000000000000fb0000000800540069006d00650100000000000004500000000000000000000004770000036c00000004000000040000000800000008fc0000000100000002000000010000000a0054006f006f006c00730000000000ffffffff0000000000000000
   Width: 1544
   X: 72
   Y: 27

